packages/reports/addon/components/common-actions/schedule.ts

```diff
@@ -249,6 +256,24 @@ export default class ScheduleActionComponent extends Component<Args> {
     this.localDeliveryRule.format.type = type;
   }
 
+  @action
+  toggleOverwriteFile() {
+    assert('The localDeliveryRule is defined', this.localDeliveryRule);
+    assert(
+      'Must be a type that supports overwriteFile in order to toggle',
+      this.overwriteableFormats.includes(this.localDeliveryRule.format.type)
+    );
+    const format = this.localDeliveryRule.format as GsheetFormat; //currently only type that supports this, make more dynamic with new types
```

Comment:
Looked into the typing around this a little bit and I think the reason visualizations work but this doesn't when using the type discriminator is that visualizations are typed via a mixin

https://github.com/yavin-dev/framework/blob/ee70aecb7e817229146d86a2b8b886d2c94051b5/packages/core/addon/mixins/models/has-visualization.ts#L15